components/sup/src/manager/mod.rs

```diff
@@ -613,6 +630,53 @@ impl Suitability for SuitabilityLookup {
     }
 }
 
+fn obtain_process_lock(fs_cfg: &FsCfg) -> Result<()> {
+    match write_process_lock(&fs_cfg.proc_lock_file) {
+        Ok(()) => Ok(()),
+        Err(_) => {
+            match File::open(&fs_cfg.proc_lock_file) {
+                Ok(file) => {
+                    let reader = BufReader::new(file);
+                    if let Some(Ok(line)) = reader.lines().next() {
+                        if let Ok(pid) = line.parse::<u32>() {
+                            if process::is_alive(pid) {
+                                return Err(sup_error!(Error::ProcessLocked(pid)));
+                            }
+                        }
+                    }
+                    release_process_lock(fs_cfg);
```

Comment:
@adamhjk technically by doing this there is a race condition where another supervisor of the same name is simultaneously attempting to obtain a process lock. It'll be exceedingly rare, but if one process hits this function while another just did this check and exited the function call below, we could still end up with two supervisors.

I think it's alright to do this for now but what do you think of using `flock` (and the Windows counterpart if existing)  instead in a future improvement?